src/studio/ui/QGrid.java

```diff
@@ -35,48 +35,27 @@ public int getRowCount() {
     static class MYJTable extends JTable {
         public MYJTable(TableModel m) {
             super(m);
+            Font tableFont = Config.getInstance().getFont(Config.FontKind.TABLE);
+            setFont(tableFont);
```

Comment:
Column width in the QGrid is incorrectly calculated (I set very big font and got ... instead of cell content). The problem is likely in WidthAdjuster

Reply:
Can't reproduce (Arch linux, openjdk 17). Since I don't have access to a real KDB I use this trick:
```diff
diff --git a/src/studio/ui/StudioPanel.java b/src/studio/ui/StudioPanel.java
index 1d12b03..c07b87d 100755
--- a/src/studio/ui/StudioPanel.java
+++ b/src/studio/ui/StudioPanel.java
@@ -542,10 +542,13 @@ public class StudioPanel extends JPanel implements Observer,WindowListener {
     }
 
     private void openFile() {
-        String filename = chooseFilename();
-        if (filename == null) return;
-        addToMruFiles(filename);
-        addTab(editor.getServer(), filename);
+        QueryResult queryResult = new QueryResult(new Server(), "aaa");
+        queryResult.setResult(new K.KSymbolVector("aaa", "bbb", "ccc", "dddddddddddddd dddddddddddddd ddeee11"));
+        queryExecutionComplete(getEditor(0), queryResult);
     }
 
     public void loadMRUFile(String filename) {

```

Reply:
I am on MacOS, Java 8, tried both Metal and Nimbus L&F. I got consistent behaviour on your use-case (see below). Interesting that if I double click on header vertical line (to trigger WithAdjuster.resize() ), the width is calculated correctly.

Aslo I don't understand how it assumed to be working. Debugging of WidthAdjuster, I got in line 90, a component with Font of Monospaced size=12 (while I set size=14 in the config). And I think this is a header renderer - see assignment in the line 73. But what I didn't yet get to the bottom, why I got correct width when I trigger resize again.

![image](https://user-images.githubusercontent.com/29353814/134053937-4d92a3f2-686a-46a6-8ae4-95438db206b7.png)